### PR TITLE
bcda-4219 Feature: support kcc and dc models

### PR DESCRIPTION
**Pull request by SMLuthi**
### Fixes [BCDA-4219](https://jira.cms.gov/browse/BCDA-4219)

Update the api to allow for kcc (ckcc/kcf) and dc cclf models.

### Proposed Changes

Update our models service to allow for the ckcc, kcf, and dc cms id types. Update the filename regex in `fileprocessor.go` to not only allow for the new file types but to tighten up our regexing in general and ONLY allow for the specific file types. Update our cclf synthetic file generation to create new dev files for ckcc, kcf, and dc models.

### Change Details

- Updated `models/service.go` to allow new kcc and dc models by adding the cms_id to regex.
- Updated `fileprocessor.go` regex to specifically check for our currently whitelisted aco types.
- Updated `cclfUtils.go` to generate synthetic data for the new aco types (kcc and dc).
- Updated makefile `load-synthetic-cclf-data` and add dev generation for kcc and dc models.

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

The API now accepts the new kcc and dc aco model types.

We generate synthetic data for the new model types.

Screen Shot 2021-02-18 at 4 03 16 PM

Screen Shot 2021-02-18 at 4 03 35 PM

Screen Shot 2021-02-18 at 4 03 48 PM

### Feedback Requested

Usual feedback; let me know if this could benefit from more testing and let me know if Go improvements could be done.

### CODE REVIEW

File: bcda/cclf/fileprocessor.go [hunk 1]
```diff
@@ -135,10 +135,30 @@ func getCMSID(name string) (string, error) {
 
 func getCCLFFileMetadata(cmsID, fileName string) (cclfFileMetadata, error) {
 	var metadata cclfFileMetadata
-	// CCLF filename convention for SSP with BCD identifier: P.BCD.A****.ZC[0|8][Y|R]**.Dyymmdd.Thhmmsst
-	// CCLF filename convention for NGACO:  P.V***.ACO.ZC[0|8][Y|R].Dyymmdd.Thhmmsst
-	// CCLF file name convetion for CEC: P.CEC.ZC[0|8][Y|R].Dyymmdd.Thhmmsst
-	filenameRegexp := regexp.MustCompile(`(T|P)(?:\.BCD)?\.(.*?)(?:\.ACO)?\.ZC(0|8)(Y|R)(\d{2})\.(D\d{6}\.T\d{6})\d`)
+	const (
```
Comment (adamayd) [inline bcda/cclf/fileprocessor.go:138]: This is awesomesauce * 5000

**Review by adamayd — APPROVED** [review]
LGTM!!

File: bcda/cclf/testutils/cclfUtils.go [hunk 2]
```diff
@@ -160,3 +166,12 @@ func addFileToZip(zipWriter *zip.Writer, filename string) error {
 	_, err = io.Copy(writer, fileToZip)
 	return err
 }
+
+func hasAnyPrefix(s string, prefixes ...string) bool {
```
Comment (mtrang1263) [inline bcda/cclf/testutils/cclfUtils.go:170]: Nice!

**Review by mtrang1263 — APPROVED** [review]
LGTM!

**Review by CosmicSyntax — APPROVED** [review]
LGTM!

**Review by katyasoup — APPROVED** [review]
LGTM!

**Pull request merged**